### PR TITLE
docs: add changelog

### PR DESCRIPTION
**Pull request by mr-cal**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

Prep for `2.9.0` release.

Adds the changelog to the documentation per [starbase](https://github.com/canonical/starbase/blob/main/HACKING.rst#changelog) conventions.

Preview here: https://canonical-craft-application--366.com.readthedocs.build/en/366/reference/changelog.html

### CODE REVIEW

File: docs/reference/changelog.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+-------------------
+
+Base naming convention
+======================
+
+Applications that use a non-default base naming convention must implement
+``Project._providers_base()`` to translate application-specific base names into
+a Craft Providers base.
+
+The default base naming convention is ``<name>@<channel>``. For example,
```
Comment (dariuszd21) [inline docs/reference/changelog.rst:58]: Is it `<name>@<channel>` or `<distro>@<series>`?

**Review by dariuszd21 — APPROVED** [review]
Looks good, thank!

File: docs/reference/changelog.rst (hunk 1 above)
Comment (mr-cal) [reply docs/reference/changelog.rst:58]: Good catch, I forgot we agreed to that in craft-platforms.

File: docs/reference/changelog.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+* Fixes a bug where the pack command could accept a list of parts as command
+  line arguments.
+* Adds support for commands to accept multiple ``platform`` or ``build-for``
+  values from the command line as comma-separated values.
+
+Remote build
+============
+
+* Retries more API calls to Launchpad.
+* Adds an exponentially backoff to API retries with a maximum total delay of
```
Comment (mattculler) [inline docs/reference/changelog.rst:54]:
```suggestion
* Adds an exponential backoff to API retries with a maximum total delay of
```

File: docs/reference/changelog.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  62 seconds.
```
Comment (mattculler) [inline docs/reference/changelog.rst:55]: Is 62 seconds really the correct value?

File: docs/reference/changelog.rst [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+* Adds a default ``Platform`` model. See :doc:`platforms</reference/platforms>`
+  for a reference of the model.
+* Adds a default ``get_build_plan()`` function to the ``BuildPlanner`` class.
+  See :doc:`Build plans</explanation/build-plans>` for an explanation of how
+  the default ``get_build_plan()`` works.
+* Changes ``BuildPlanner`` from an abstract class to a fully implemented class.
+  Applications can now use the ``BuildPlanner`` class directly.
+
+For a complete list of commits, check out the `2.9.0`_ release on GitHub.
```
Comment (mattculler) [inline docs/reference/changelog.rst:37]:
```suggestion
For a complete list of commits, check out the `3.0.0`_ release on GitHub.
```
^ I assume

**Review by mattculler — APPROVED** [review]
Excellent

File: docs/reference/changelog.rst (hunk 3 above)
Comment (mr-cal) [reply docs/reference/changelog.rst:55]: It does seem odd but yes: 2+4+8+16+32 = 62 seconds

https://github.com/canonical/craft-application/commit/b97d76d62d0c48ce216db37376353401508895c4

File: docs/reference/changelog.rst (hunk 4 above)
Comment (mr-cal) [reply docs/reference/changelog.rst:37]: Ah, thanks!

**Pull request merged**